dffml/operation/output.py

```diff
@@ -184,6 +191,12 @@ async def run(self, inputs: Dict[str, Any]) -> Dict[str, Any]:
                 async for item in od.inputs(definition):
                     want.setdefault(definition.name, [])
                     want[definition.name].append(item.value)
+
+            # Rename outputs if present in name_map
+            for key, value in want.items():
+                if name_map.get(key, None):
+                    want[name_map[key]] = value
+                    want.pop(key)
```

Comment:
```
2020-07-20T10:19:28.1178646Z ======================================================================
2020-07-20T10:19:28.1178800Z ERROR: test_docstring (tests.test_docstrings.operation_output_GetSingle)
2020-07-20T10:19:28.1179782Z ----------------------------------------------------------------------
2020-07-20T10:19:28.1179927Z Traceback (most recent call last):
2020-07-20T10:19:28.1180484Z   File "/home/runner/work/dffml/dffml/tests/test_docstrings.py", line 209, in testcase
2020-07-20T10:19:28.1180684Z     run_doctest(obj, state)
2020-07-20T10:19:28.1181014Z   File "/home/runner/work/dffml/dffml/tests/test_docstrings.py", line 178, in run_doctest
2020-07-20T10:19:28.1181213Z     raise Exception(output.getvalue())
2020-07-20T10:19:28.1181343Z Exception: Trying:
2020-07-20T10:19:28.1181469Z     import asyncio
2020-07-20T10:19:28.1181590Z Expecting nothing
2020-07-20T10:19:28.1181715Z ok
2020-07-20T10:19:28.1181833Z Trying:
2020-07-20T10:19:28.1181958Z     from dffml import *
2020-07-20T10:19:28.1182080Z Expecting nothing
2020-07-20T10:19:28.1182200Z ok
2020-07-20T10:19:28.1182304Z Trying:
2020-07-20T10:19:28.1182438Z     URL = Definition(name="URL", primitive="string")
2020-07-20T10:19:28.1182573Z Expecting nothing
2020-07-20T10:19:28.1182691Z ok
2020-07-20T10:19:28.1182808Z Trying:
2020-07-20T10:19:28.1182936Z     ORG = Definition(name="ORG", primitive="string")
2020-07-20T10:19:28.1183064Z Expecting nothing
2020-07-20T10:19:28.1183182Z ok
2020-07-20T10:19:28.1183298Z Trying:
2020-07-20T10:19:28.1183428Z     dataflow = DataFlow.auto(GetSingle)
2020-07-20T10:19:28.1183552Z Expecting nothing
2020-07-20T10:19:28.1183672Z ok
2020-07-20T10:19:28.1183789Z Trying:
2020-07-20T10:19:28.1183912Z     dataflow.seed.append(
2020-07-20T10:19:28.1184041Z         Input(
2020-07-20T10:19:28.1184181Z             value=[{"Repo Link": URL.name}, ORG.name],
2020-07-20T10:19:28.1184329Z             definition=GetSingle.op.inputs["spec"]
2020-07-20T10:19:28.1184463Z         )
2020-07-20T10:19:28.1184584Z     )
2020-07-20T10:19:28.1184701Z Expecting nothing
2020-07-20T10:19:28.1184807Z ok
2020-07-20T10:19:28.1184925Z Trying:
2020-07-20T10:19:28.1185046Z     async def main():
2020-07-20T10:19:28.1185191Z         async for ctx, results in MemoryOrchestrator.run(dataflow, [
2020-07-20T10:19:28.1185335Z             Input(
2020-07-20T10:19:28.1185509Z                 value="https://github.com/intel/dffml",
2020-07-20T10:19:28.1185666Z                 definition=URL
2020-07-20T10:19:28.1185798Z             ),
2020-07-20T10:19:28.1185930Z             Input(
2020-07-20T10:19:28.1186064Z                 value="Intel",
2020-07-20T10:19:28.1186200Z                 definition=ORG
2020-07-20T10:19:28.1186329Z             )
2020-07-20T10:19:28.1186457Z         ]):
2020-07-20T10:19:28.1186587Z             print(results)
2020-07-20T10:19:28.1186710Z Expecting nothing
2020-07-20T10:19:28.1186829Z ok
2020-07-20T10:19:28.1186944Z Trying:
2020-07-20T10:19:28.1187065Z     asyncio.run(main())
2020-07-20T10:19:28.1187186Z Expecting:
2020-07-20T10:19:28.1187596Z     {'ORG': 'Intel', 'Repo Link': 'https://github.com/intel/dffml'}
2020-07-20T10:19:28.1187761Z **********************************************************************
2020-07-20T10:19:28.1188138Z File "/home/runner/work/dffml/dffml/dffml/operation/output.py", line 263, in GetSingle
2020-07-20T10:19:28.1188658Z Failed example:
2020-07-20T10:19:28.1188793Z     asyncio.run(main())
2020-07-20T10:19:28.1188915Z Exception raised:
2020-07-20T10:19:28.1189046Z     Traceback (most recent call last):
2020-07-20T10:19:28.1189526Z       File "/usr/share/miniconda/lib/python3.8/doctest.py", line 1336, in __run
2020-07-20T10:19:28.1189734Z         exec(compile(example.source, filename, "single",
2020-07-20T10:19:28.1190244Z       File "<doctest GetSingle[7]>", line 1, in <module>
2020-07-20T10:19:28.1190413Z         asyncio.run(main())
2020-07-20T10:19:28.1190690Z       File "/usr/share/miniconda/lib/python3.8/asyncio/runners.py", line 43, in run
2020-07-20T10:19:28.1190863Z         return loop.run_until_complete(main)
2020-07-20T10:19:28.1191158Z       File "/usr/share/miniconda/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
2020-07-20T10:19:28.1191461Z         return future.result()
2020-07-20T10:19:28.1191740Z       File "<doctest GetSingle[6]>", line 2, in main
2020-07-20T10:19:28.1191904Z         async for ctx, results in MemoryOrchestrator.run(dataflow, [
2020-07-20T10:19:28.1192219Z       File "/home/runner/work/dffml/dffml/dffml/df/base.py", line 1025, in run
2020-07-20T10:19:28.1192402Z         async for ctx, results in octx.run(inputs):
2020-07-20T10:19:28.1192656Z       File "/home/runner/work/dffml/dffml/dffml/df/memory.py", line 1426, in run
2020-07-20T10:19:28.1192832Z         raise exception
2020-07-20T10:19:28.1193100Z       File "/home/runner/work/dffml/dffml/dffml/df/memory.py", line 1633, in run_operations_for_ctx
2020-07-20T10:19:28.1193267Z         output = {
2020-07-20T10:19:28.1193521Z       File "/home/runner/work/dffml/dffml/dffml/df/memory.py", line 1633, in <dictcomp>
2020-07-20T10:19:28.1193687Z         output = {
2020-07-20T10:19:28.1193937Z       File "/home/runner/work/dffml/dffml/dffml/df/memory.py", line 1658, in run_stage
2020-07-20T10:19:28.1194116Z         yield operation, await self.nctx.run(
2020-07-20T10:19:28.1194420Z       File "/home/runner/work/dffml/dffml/dffml/df/memory.py", line 1025, in run
2020-07-20T10:19:28.1194589Z         outputs = await opctx.run(inputs)
2020-07-20T10:19:28.1195034Z       File "/home/runner/work/dffml/dffml/dffml/operation/output.py", line 269, in run
2020-07-20T10:19:28.1195220Z         want = await super().run(inputs)
2020-07-20T10:19:28.1195418Z       File "/home/runner/work/dffml/dffml/dffml/operation/output.py", line 196, in run
2020-07-20T10:19:28.1195584Z         for key, value in want.items():
2020-07-20T10:19:28.1195724Z     RuntimeError: dictionary keys changed during iteration
```

Reply:
@pdxjohnny Ah I must've missed it. I've pushed a fix!